lib/Sema/ConstraintSystem.cpp

```diff
@@ -1774,9 +1774,12 @@ FunctionType *ConstraintSystem::adjustFunctionTypeForConcurrency(
           adjustedTy =
               adjustedTy->withExtInfo(adjustedTy->getExtInfo().withSendable());
         }
-      } else if (isPartialApplication(getConstraintLocator(locator))) {
-        if (baseType &&
-            (baseType->is<AnyMetatypeType>() || baseType->isSendableType())) {
+      } else if (numApplies < decl->getNumCurryLevels()) {
+        // Operators on protocols could be found via unqualified lookup and
+        // won't have a base type.
+        if (decl->isOperator() ||
+            (baseType &&
+             (baseType->is<AnyMetatypeType>() || baseType->isSendableType()))) {
```

Comment:
Actually wait I think we need to check more than `baseType` for the curried instance method with applied self case, i.e for `X.foo(X())`, the base being a metatype is always true, but what matters is the Sendability of the instance type

Reply:
You are right! I'm working on a fix!

Reply:
✅